apps/dashboard/src/components/workflow-editor/steps/use-editor-preview.tsx

```diff
@@ -16,11 +17,18 @@ type UseEditorPreviewProps = {
 
 function useDebounced<T>(value: T, delay: number): T {
   const [debouncedValue, setDebouncedValue] = useState<T>(value);
+  const oldValueRef = useDataRef(debouncedValue);
 
   useEffect(() => {
-    const timer = setTimeout(() => setDebouncedValue(value), delay);
+    const timer = setTimeout(() => {
+      const oldValue = JSON.stringify(oldValueRef.current);
+      const newValue = JSON.stringify(value);
+      if (oldValue === newValue) return;
+
+      setDebouncedValue(value);
+    }, delay);
```

Comment:
this is to fix the re-rendering problem on the editor-preview context, cc @scopsy